src/views/catalog/CreateFromInstanceTypes/state/hooks/useVirtualMachineInstanceTypes.ts

```diff
@@ -3,32 +3,30 @@ import { V1beta1VirtualMachineInstancetype } from '@kubevirt-ui/kubevirt-api/kub
 import { ALL_NAMESPACES_SESSION_KEY } from '@kubevirt-utils/hooks/constants';
 import useClusterParam from '@multicluster/hooks/useClusterParam';
 import useK8sWatchData from '@multicluster/hooks/useK8sWatchData';
-import { Selector, useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import { Selector } from '@openshift-console/dynamic-plugin-sdk';
 
-type UseVirtualMachineInstanceTypes = (
-  fieldSelector?: string,
-  selector?: Selector,
-  fetchAllProjects?: boolean,
-) => [instanceTypes: V1beta1VirtualMachineInstancetype[], loaded: boolean, loadError: Error];
+type UseVirtualMachineInstanceTypes = (params?: {
+  fieldSelector?: string;
+  namespace?: string;
+  selector?: Selector;
+}) => [instanceTypes: V1beta1VirtualMachineInstancetype[], loaded: boolean, loadError: Error];
 
-const useVirtualMachineInstanceTypes: UseVirtualMachineInstanceTypes = (
+const useVirtualMachineInstanceTypes: UseVirtualMachineInstanceTypes = ({
   fieldSelector,
+  namespace,
   selector,
-  fetchAllProjects,
-) => {
-  const [activeNamespace] = useActiveNamespace();
+}) => {
   const cluster = useClusterParam();
-  const isAllNamespace = activeNamespace === ALL_NAMESPACES_SESSION_KEY;
-  const [instanceTypes, loaded, loadError] = useK8sWatchData<V1beta1VirtualMachineInstancetype[]>(
-    (fetchAllProjects || !isAllNamespace) && {
-      cluster,
-      fieldSelector,
-      groupVersionKind: VirtualMachineInstancetypeModelGroupVersionKind,
-      isList: true,
-      selector,
-      ...(!isAllNamespace && { namespace: activeNamespace }),
-    },
-  );
+  const isAllNamespace = namespace === ALL_NAMESPACES_SESSION_KEY;
```

Comment:
We have a utility function named `isAllNamespaces` for this. It has more checks than this, so make sure that won't cause a problem.